packages/builder/src/components/backend/Datasources/CreateEditRelationship.svelte

```diff
@@ -271,6 +277,39 @@
     toRelationship = relateTo
   }
 
+  function relationshipExists() {
+    if (
+      originalFromTable &&
+      originalToTable &&
+      originalFromTable === fromTable &&
+      originalToTable === toTable
+    ) {
+      return false
+    }
+    let match = false
+    let fromThroughLinks = Object.values(
+      datasource.entities[fromTable.name].schema
+    ).filter(value => value.through)
+    let toThroughLinks = Object.values(
+      datasource.entities[toTable.name].schema
+    ).filter(value => value.through)
+
+    fromThroughLinks.forEach(fromVal => {
```

Comment:
Logically this is quite difficult to follow, wondering if something like:
```js
const compareAgainstUserInput = (fromTableId, toTableId) => (fromTableId === fromId && toTableId === toId) || (fromTableId === toId && toTableId === fromId)
fromThroughLinks.find(from => 
  toThroughLinks.find(to => from.through === to.through && compareAgainstUserInput(from.tableId, to.tableId)))
```

Might need some formatting, but it breaks it up a little better and remove the double looping which isn't technically exited fully (the inner `toThroughLinks` is stopped, but the external `fromThroughLinks` will keep iterating currently which could cause issues down the line if this function was ever updated).

Reply:
That does look a bit cleaner - thanks!